### PR TITLE
Slinky: use LEDs when available

### PR DESCRIPTION
**Pull request by nkaje**
1. apps/slinky: use led when available
2. slinky: enable console history

### CODE REVIEW

File: apps/slinky/src/led.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+STATS_SECT_END
+
+static STATS_NAME_START(gpio_stats)
+STATS_NAME(gpio_stats, toggles)
+STATS_NAME_END(gpio_stats)
+
+static STATS_SECT_DECL(gpio_stats) g_stats_gpio_toggle;
+
+void
+init_led_stats()
```
Comment (kasjer) [inline apps/slinky/src/led.c]: missing void in ()

File: apps/slinky/src/led.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    stats_init(STATS_HDR(g_stats_gpio_toggle),
+               STATS_SIZE_INIT_PARMS(g_stats_gpio_toggle, STATS_SIZE_32),
+               STATS_NAME_INIT_PARMS(gpio_stats));
+
+    stats_register("gpio_toggle", STATS_HDR(g_stats_gpio_toggle));
+}
+
+void
+toggle_led()
```
Comment (kasjer) [inline apps/slinky/src/led.c]: missing void in ()

File: apps/slinky/src/led.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    DFLT_LOG_INFO("GPIO toggle from %u to %u",
+                    prev_pin_state, curr_pin_state);
+    STATS_INC(g_stats_gpio_toggle, toggles);
+}
+
+#else
+void
+init_led_stats()
+{
+    return;
```
Comment (kasjer) [inline apps/slinky/src/led.c]: return is not needed here

File: apps/slinky/src/led.c [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    prev_pin_state = hal_gpio_read(g_led_pin);
+    curr_pin_state = hal_gpio_toggle(g_led_pin);
+    DFLT_LOG_INFO("GPIO toggle from %u to %u",
+                    prev_pin_state, curr_pin_state);
+    STATS_INC(g_stats_gpio_toggle, toggles);
+}
+
+#else
+void
+init_led_stats()
```
Comment (kasjer) [inline apps/slinky/src/led.c]: void missing in ()

File: apps/slinky/src/led.c [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+               STATS_SIZE_INIT_PARMS(g_stats_gpio_toggle, STATS_SIZE_32),
+               STATS_NAME_INIT_PARMS(gpio_stats));
+
+    stats_register("gpio_toggle", STATS_HDR(g_stats_gpio_toggle));
+}
+
+void
+toggle_led()
+{
+    static int prev_pin_state, curr_pin_state;
```
Comment (kasjer) [inline apps/slinky/src/led.c]: static here does not add any value

File: apps/slinky/src/led.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    g_led_pin = LED_BLINK_PIN;
```
Comment (kasjer) [inline apps/slinky/src/led.c]: setting g_led_pin here does not seem needed here, it was already set in init_led_stats.
In fact variable could be skipped altogether.

File: apps/slinky/src/led.c (hunk 1 above)
Comment (nkaje) [reply apps/slinky/src/led.c]: done

File: apps/slinky/src/led.c (hunk 2 above)
Comment (nkaje) [reply apps/slinky/src/led.c]: done

File: apps/slinky/src/led.c (hunk 3 above)
Comment (nkaje) [reply apps/slinky/src/led.c]: removed

File: apps/slinky/src/led.c (hunk 4 above)
Comment (nkaje) [reply apps/slinky/src/led.c]: removed

File: apps/slinky/src/led.c (hunk 5 above)
Comment (nkaje) [reply apps/slinky/src/led.c]: removed

File: apps/slinky/src/led.c (hunk 6 above)
Comment (nkaje) [reply apps/slinky/src/led.c]: removed

**Pull request merged**